From: Christoph Hellwig <hch@lst.de>
Date: Tue, 19 Jul 2016 11:23:34 +0200
Subject: block: don't ignore -EOPNOTSUPP blkdev_issue_write_same
References: bnc#1023798,FATE#321463
Patch-Mainline: v4.8-rc1
Git-commit: 3f40bf2c89223192535a72a4d49ce7d68d8ed9dd

WRITE SAME is a data integrity operation and we can't simply ignore
errors.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Martin K. Petersen <martin.petersen@oracle.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-lib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/blk-lib.c b/block/blk-lib.c
index 7c3c6c0..083e56f 100644
--- a/block/blk-lib.c
+++ b/block/blk-lib.c
@@ -181,7 +181,7 @@ int blkdev_issue_write_same(struct block_device *bdev, sector_t sector,
 		ret = submit_bio_wait(bio);
 		bio_put(bio);
 	}
-	return ret != -EOPNOTSUPP ? ret : 0;
+	return ret;
 }
 EXPORT_SYMBOL(blkdev_issue_write_same);
 
-- 
1.8.5.6

